### DISCUSSION TITLE
Avalonia XAML Previewer on Linux (alpha release)

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#7578] [General]
I'd like to share the initial release of an Avalonia Previewer. Here's a screenshot:

![image](https://user-images.githubusercontent.com/17156238/153419225-ad60e0f1-f035-4cd4-b0f9-ec09500df0c4.png)

The initial release is for Linux and there are AppImages for download on Github page:

https://github.com/kuiperzone/AvantGarde

@kekekeks I followed your suggestion about using the preview host. It was the correct way to do it in the end.

Note that this is a standalone previewer app, and not an IDE extension. There are some notes on the Github page.

It also has some features that I'd be keen to get some feedback on it. For example, it is possible to "disable events" and "prefetch assets" prior to sending to the preview host. Not sure how useful this will be, but it came from my earlier efforts and here's the thinking:

Why would you want to disable events? Because it allows events to be added and modified and keep generating a preview without having to rebuild the assembly. The same goes for asset prefetch which will find the assets on the file system. Again, assets can be added without a rebuild.

The "preview options" are highlighted in the screenshot below:

![image](https://user-images.githubusercontent.com/17156238/153422133-d31fe883-88f1-4d42-9f29-a6533046c988.png)

If it's not clear above, the application is actually previewing itself.

It can also spit out an Avalonia XSD schema file (under the File menu) which is compatible with the "XML Complete" Visual Code extension. Screenshot below in visual code:

**Comment by danwalmsley** [comment]
WOW!!!!!!!!!

**Comment by kuiperzone** [comment]
If you want to raise issues and suggestions, that would be great. I think I will need a break for a couple of weeks before doing any major changes, but will look at them and respond.

https://github.com/kuiperzone/AvantGarde/issues

**Comment by odalet** [comment]
You said the initial release was for Linux, so I immediately downloaded the code and built it on my Windows machine and it works perfectly (well, I had to comment out the post build that copies the license: `cp` is linux-only, should be `copy` and maybe anti-slashes for this to work in Windows - but apart from this, no change).

And I prove it 😄 

![image](https://user-images.githubusercontent.com/79333/153435665-fe3da50c-06a9-4773-949c-bdd722671450.png)

**Reply by kuiperzone** [reply]
Hey, that's great news. The only reason I said it was for Linux was because I just haven't tested it properly on Windows. It's great that it works almost out of the box. Thanks for letting me know.

**Reply by kuiperzone** [reply]
Spotted a possible issue from your screenshot. It should actually show relative assembly paths, but is showing fully qualified.

![image](https://user-images.githubusercontent.com/17156238/153441281-6d18c95c-4443-420a-882a-e8e5cee102d3.png)

**Reply by odalet** [reply]
Found why: in PathItem.cs, line 187, replace `Path.PathSeparator` by `Path.DirectorySeparatorChar` and it works as you intended

**Reply by odalet** [reply]
See <https://github.com/kuiperzone/AvantGarde/pull/2>

**Comment by Anequit** [comment]
Love this!

**Comment by kuiperzone** [comment]
Have released update version:

https://github.com/kuiperzone/AvantGarde

**Comment by furai** [comment]
Great work!
